[slye20] iP

slye20 opened the pull request:
# Bob 🤓 

> "We love BOB" - Some important people ([source](https://geo-g.fandom.com/wiki/Bob))

Bob helps you to: 
- Keep tracks of your tasks for the *day*
- Manage them

What type of tasks:
1. ~~Task~~ Todos
2. Deadlines
3. Events

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
`HELLO WORLD`

**Features so far:**
- [x] Tasks
- [ ] GUI

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

slye20 committed:
Change bot name from Duke to EchoMind

slye20 committed:
Add [EchoMind] before every response

slye20 committed:
Echo commands by users

slye20 committed:
Removed whitespace from before and after message

slye20 committed:
Store text and display back when requested

slye20 committed:
Add Task class

slye20 committed:
Mark/Unmark tasks as done

slye20 committed:
Add ToDo, Event, Deadline

slye20 committed:
Automate text UI testing

slye20 committed:
Error handling

slye20 committed:
Delete task

slye20 committed:
Refactor code

slye20 committed:
Fix exception for deadline and event

slye20 committed:
Code refactoring

slye20 committed:
Save tasks in hard disk

slye20 committed:
Merge branch 'branch-level-7'

slye20 committed:
Add date

slye20 committed:
Make code more OOP

slye20 committed:
Merge branch 'add-gradle-support'

slye20 committed:
Add gradle

slye20 committed:
Add JUnit tests

slye20 committed:
Create JAR file

slye20 committed:
Add JavaDoc

slye20 committed:
Edit Coding Standard

slye20 committed:
Add find task feature

slye20 committed:
Add messge when cannot find

slye20 committed:
Resolve merge conflict with coding standaard

slye20 committed:
Resolve merge conflict with level-9

JYL27 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,120 @@
+     * @param t The {@code TaskList} where the task will be marked as done.
+     * @throws BobException If the message format is invalid.
+     */
+    public void markTaskParser(String message, TaskList t) throws BobException {
```
The formatting of the names of the methods could be somewhat misleading. This method in particular could sound like you are marking a task parser. Perhaps something like parseMarkTaskCommand instead?

I've noticed the same thing in the Ui class as well.

JYL27 commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,94 @@
+    /**
+     * Displays the introductory messages when the application starts.
+     */
+    public void introBobUi() {
```
"intro" could come off as somewhat slangy. Maybe just introduceBobUi instead?

matthewyip1511 replied:
I agree, `introduce` would be clearer in this context

JYL27 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,97 @@
+     *
+     * @param tl The {@code TaskList} where tasks will be loaded into.
+     */
+    public void loadTasks(TaskList tl) {
```
I like the method names here, very clear and concise 👍

JYL27 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,97 @@
+            while (s1.hasNext()) {
+                String t = s1.nextLine();
+                String[] task_list = t.trim().split(" \\| ");
+                Task x;
```
I like the general consistency of your abbreviations. This particular Task object is abbreviated as x instead of t like in other places though. Any reason why the String object is abbreviated as t and the Task object is abbreviated as x, rather than abbreviating the task object as t and using another abbreviation for the String object?

JYL27 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,97 @@
+            Scanner s1 = new Scanner(f);
+            while (s1.hasNext()) {
+                String t = s1.nextLine();
+                String[] task_list = t.trim().split(" \\| ");
```
Not sure I like the usage of snake case here. Any reason why this variable in particular is in snake case instead of camel case?

JYL27 reviewed (commented):
LGTM👍. Just a few nits regarding variable and method naming.

slye20 committed:
Use CheckStyle

matthewyip1511 commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,37 @@
+     */
+    @Override
+    public String toSave() {
+        return "E" + " | " + (this.isDone ? 1 : 0) + " | " + this.description + " | " + this.start + " | " + this.end;
```
This line is a bit long, you might want to add a line break here 🙂

matthewyip1511 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+     *
+     * @param num The index of the task to be marked as done (0-based index).
+     */
+    public void markTask(int num) {
```
this might be better named as `markTaskAsDone` to be more descriptive 🙂

matthewyip1511 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+     *
+     * @param num The index of the task to be unmarked as not done (0-based index).
+     */
+    public void unmarkTask(int num) {
```
similarly this could be `unmarkTaskAsDone` or `markTaskAsUndone`

matthewyip1511 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,120 @@
+        if (!message.matches("^todo.*")) {
+            throw new BobException("Invalid format");
+        }
+        String x = message.replaceFirst("todo ", "");
```
The variable name should be more descriptive here 🙂 , instead of `x` it could be `parsedMessage`

matthewyip1511 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    /**
+     * Get status icon
```
Nitpicking here, but it should be `Gets` rather than `Get`. Otherwise most of the javadoc comments are in line with convention, good job!

matthewyip1511 commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,103 @@
+                this.parser.addEventParser(message, this.tasks);
+                break;
+            case "find":
+                this.parser.findParser(message, this.tasks);
```
Function names here are clear and easy to follow 👍

matthewyip1511 reviewed (commented):
Coding standards are mostly met overall! Just a few nits here and there, but otherwise good work so far 👍

slye20 committed:
Add GUI with javaFX

slye20 committed:
Fix exit app

slye20 committed:
Update unit test

slye20 committed:
Add name to window

slye20 committed:
Review pr comments

slye20 committed:
Add logic and assertion to check for invalid indexes

Handling invalid index is crucial to check for input errors.

Let's,
* Enable assertions in build.gradle.
* Created a validIndex function in Parser to check if index is valid.
* Add assertion in TaskList to makesure that the validIndex function works.

slye20 committed:
Add assertions in TaskList

Assert valid inputs to add functions and saveTask and findTasks

Let's check for valid decription, start and end times, and checks in saveTasks and findTasks.

slye20 committed:
Add assert to Storage.java

Assert to ensure that assumptions in task loading and saving are valid

let's add assertions to the functions: LoadTasks, prepareSave, saveTask

slye20 committed:
Add assertions in Parser.java

slye20 committed:
Refactor code: SLAP and magic numbers

slye20 committed:
Add stream to find task function

slye20 committed:
Merge pull request #2 from slye20/branch-A-Assertions

Add assertions

slye20 committed:
Fix merge conflict with master

slye20 committed:
Merge pull request #3 from slye20/branch-A-CodeQuality

Refactor code: SLAP and magic numbers

slye20 committed:
Resolve merge conflix with streams

slye20 committed:
Merge pull request #4 from slye20/branch-A-Streams

Add stream to find task function

slye20 committed:
Add feature to detect duplicate

slye20 committed:
Improve UI

slye20 committed:
Add more unit testing

slye20 committed:
Fix bug relating to text truncation

slye20 committed:
Merge branch 'branch-A-BetterGui'

slye20 committed:
Add Ui.png

slye20 committed:
Edit README.md

slye20 committed:
Add and update JavaDocs to classes and methods

slye20 committed:
Edit README.md

slye20 committed:
Refactor code

slye20 committed:
Fix JavaFX configuration warning

slye20 committed:
Edit README.md to specify date format

slye20 committed:
Fix typo in README.md

slye20 committed:
Fix more typos in README.md

slye20 committed:
Fix typo in README.md
